### DISCUSSION TITLE
Converting unstructured data to structured data

### DISCUSSION THREAD

**Discussion by anskwagh** [RenderKit/ospray#581] [Q&A]
Currently ustructured volume data is performing very poorly as compared to structured regular approach.
Is there any way that unstructured data will be automatically converted/interpolated using some algorithm and rendered using structured regular volume? If ospray could do that interanally it might lead to significant performance improvement while working on models at runtime. While exporting image/video we can use unstructured data for quality .

**Comment by johguenther** [comment]
It is always a question which feature to support behind the OSPRay API and what can be done on the application side. I looks like that besides convenience there is no compelling reason to resample the data internally (like special knowledge which an application cannot exploit). Using Open VKL to resample an Unstructured Volume into a regular grid should be easy.
